### PR TITLE
Update pyyaml and flex contraints

### PR DESCRIPTION
**Pull request by blag**
Update requirements files with updated PyYAML and flex constraints.

I also add a make target to run `make requirements` and check that it doesn't change any requirements files (eg: some files were changed but not checked into git), and integrated that into the Travis tests.

H/T: @m4dcoder

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ argcomplete
 bcrypt
 cryptography==2.4.2
 eventlet==0.24.1
-flex>=6.13.2,<6.15.0
+flex
```
Comment (m4dcoder) [inline requirements.txt]: In your other PR at https://github.com/StackStorm/st2/pull/4510, `flex` is pinned to `flex>=6.13.2,<6.15.0`.  How come the version specific dependencies are gone?

File: Makefile [hunk 2]
```diff
@@ -132,7 +132,7 @@ play:
 	@echo
 
 .PHONY: check
-check: requirements flake8 checklogs
+check: requirements flake8 checkrequirements checklogs
```
Comment (m4dcoder) [inline Makefile]: This is a good add. Thanks.

File: requirements.txt (hunk 1 above)
Comment (Kami) [reply requirements.txt]: We are missing an entry for flex in ``fixed-requirements.txt``.

I will go ahead and add it.

Comment (Kami) [reply requirements.txt]: Should be good to go now - d29cd7baa.

**Comment by Kami** [comment]
LGTM, thanks for adding new ``check-requirements`` make task :+1:

**Pull request merged**